mod/display.php

```diff
@@ -76,7 +77,7 @@ function display_init(App $a)
 	}
 
 	if (!DBA::isResult($item)) {
-		System::httpExit(404);
+		return;
```

Comment:
Why did you remove this exception?

Reply:
Any exception thrown in the `init()` method would produce a non-styled error page, which avoiding them was the initial goal of this PR.

The item availability is re-checked in the `display_content()` function anyway, and then a third time in the same function when we finally factor in the logged in user permissions.
I started refactoring the display module in `src/Module` but I had to stop because of how byzantine the logic is.